public/app/features/dashboard-scene/embedding/EmbeddedDashboard.tsx

```diff
@@ -0,0 +1,139 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { EmbeddedDashboardProps } from '@grafana/runtime';
+import { Spinner, Alert, useStyles2, Drawer, Box, TextLink, Button, Stack } from '@grafana/ui';
+
+import { getDashboardScenePageStateManager } from '../pages/DashboardScenePageStateManager';
+import { DashboardScene } from '../scene/DashboardScene';
+
+export function EmbeddedDashboard(props: EmbeddedDashboardProps) {
+  const stateManager = getDashboardScenePageStateManager();
+  const { dashboard, loadError } = stateManager.useState();
+
+  useEffect(() => {
+    stateManager.loadDashboard({ uid: props.uid!, isEmbedded: true });
+    return () => {
+      stateManager.clearState();
+    };
+  }, [stateManager, props.uid]);
+
+  if (loadError) {
+    return (
+      <Alert severity="error" title="Failed to load dashboard">
+        {loadError}
+      </Alert>
+    );
+  }
+
+  if (!dashboard) {
+    return <Spinner />;
+  }
+
+  return <EmbeddedDashboardRenderer model={dashboard} {...props} />;
+}
+
+interface RendererProps extends EmbeddedDashboardProps {
+  model: DashboardScene;
+}
+
+function EmbeddedDashboardRenderer({ model, uid, inDrawer, onClose = () => {} }: RendererProps) {
+  const [isActive, setIsActive] = useState(false);
+  const { controls, body } = model.useState();
+  const styles = useStyles2(getStyles);
+
+  useEffect(() => {
+    setIsActive(true);
+    return model.activate();
+  }, [model]);
+
+  if (!isActive) {
+    return null;
+  }
+
+  if (!inDrawer) {
+    return (
+      <div className={styles.canvas}>
+        {controls && (
+          <div className={styles.controls}>
+            {controls.map((control) => (
+              <control.Component key={control.state.key} model={control} />
+            ))}
+          </div>
+        )}
+        <div className={styles.body}>
+          <body.Component model={body} />
+        </div>
+      </div>
+    );
+  }
+
+  const title = (
+    <Stack direction="column" gap={0}>
+      <div className={styles.drawerHeader}>
+        <div className={styles.drawerTitle}>{model.state.title}</div>
+        <div className={styles.actions}>
+          <Button icon="times" variant="secondary" fill="text" onClick={onClose} />
+        </div>
+      </div>
+      <Box paddingX={2}>
+        <TextLink href={`/d/${uid}`}>Open dashboard</TextLink>
+        {controls && (
+          <div className={styles.controls}>
+            {controls.map((control) => (
+              <control.Component key={control.state.key} model={control} />
+            ))}
+          </div>
+        )}
+      </Box>
```

Comment:
This will render dash controls in the drawer title? Not sure about the space there, for scenerios with variables + links + timepicker.

Reply:
there is plenty of space, (no space restriction) and the nice thing about having it there is that time controls stays fixed when scrolling. 

But might remove this inDrawer option and just start with a plain embedded dashboard for now, Asserts needs a drawer with multiple tabs each with a separate dashboard.